### PR TITLE
feat(*): error navigation - I61

### PR DESCRIPTION
**Pull request by jolanglinais**
# Issue #61 
<OVERALL SUMMARY OF PULL REQUEST>

### Changes
- Organize related codebase
- Get latest templates from library
- Implement logic for navigating the HTML DOM and Slate DOM for errors

### Flags
- @dselman please review the `scrollToClause` functionality
  - The decrement to the node's key seems unnecessary, I may be missing something
- Draft until new release of Cicero UI is released

### Related Issues
- [Cicero UI Issue 17](https://github.com/accordproject/cicero-ui/issues/17)
- [Cicero UI PR 98](https://github.com/accordproject/cicero-ui/pull/98)

### CODE REVIEW

File: src/sagas/templatesSaga.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ export function* pushTemplatesToStore() {
   try {
     const templateLibrary = new TemplateLibrary();
     const templateIndex = yield templateLibrary
-      .getTemplateIndex({ latestVersion: false, ciceroVersion });
+      .getTemplateIndex({ latestVersion: true, ciceroVersion });
```
Comment (DianaLease) [inline src/sagas/templatesSaga.js:24]: I'm not sure I understand this change. Do we want the latest version of the templates or the versions that match the cicero version provided? Will this only give us templates if the latest version matches the passed ciceroVersion?

Comment (jolanglinais) [reply src/sagas/templatesSaga.js:24]: I am a little unclear on this. The new changes of `displayName` were not populating without this, but that was before I had updated `cicero-ui`.

I believe the `ciceroVersion` in `{ latestVersion: false, ciceroVersion }` filters for templates that match the cicero version provided from the node modules dependencies. Setting the `latestVersion` back to `false` results in two different versions of every template populating, as shown here: 

Screen Shot 2019-08-07 at 8 43 26 AM

Comment (jeromesimeon) [reply src/sagas/templatesSaga.js:24]: Template Studio v1 uses: `templateLibrary.getTemplateIndex({ latestVersion: true, ciceroVersion });` and I thought it was for a good reason, but I can't recall what that was now...

**Review by jeromesimeon — DISMISSED** [review]
LGTM

File: src/sagas/templatesSaga.js (hunk 1 above)
Comment (jolanglinais) [reply src/sagas/templatesSaga.js:24]: This works for me. I think it's an easy change in the future.

**Review by DianaLease — APPROVED** [review]
LGTM

**Pull request merged**